src/Type/Doctrine/Query/QueryResultTypeWalker.php

```diff
@@ -175,6 +179,7 @@ public function walkSelectStatement(AST\SelectStatement $AST): string
 		$this->typeBuilder->setSelectQuery();
 		$this->hasAggregateFunction = $this->hasAggregateFunction($AST);
 		$this->hasGroupByClause = $AST->groupByClause !== null;
+		$this->hasWhereClause = $AST->whereClause !== null;
```

Comment:
Are you sure this ever works? Because `QueryBuilderGetQueryDynamicReturnTypeExtension::METHODS_NOT_AFFECTING_RESULT_TYPE` contains all where methods.

And if you remove those, you will break codebases using dynamic approaches in those methods (if those have `doctrine.reportDynamicQueryBuilders: true`).